test/elixir_analyzer/exercise_test/feature/pipes_test.exs

```diff
@@ -0,0 +1,184 @@
+defmodule ElixirAnalyzer.ExerciseTest.Feature.PipesTest do
+  use ElixirAnalyzer.ExerciseTestCase,
+    exercise_test_module: ElixirAnalyzer.Support.AnalyzerVerification.Feature.Pipes
+
+  test_exercise_analysis "function with one parameter",
+    comments_exclude: ["not one parameter"] do
+    [
+      defmodule MyModule do
+        def hi(name) do
+          foo(name)
+        end
+      end,
+      # This case matches, because feature looks down the AST
+      defmodule MyModule do
+        def hi(name) do
+          name |> foo(:ok)
+        end
+      end,
```

Comment:
Do we have any tests that depend on this behavior?

Reply:
I don't think so. Not in the ones I made anyway. 

This would be an issue if you were trying to match a function of arity 1, you could get a false positive that matches the same function of arity 2 used with a pipe. I have no idea how likely that is to happen organically. To avoid that you would have say match `foo(_ignore)` but not `_ignore  |> foo(_ignore)`.

I cannot think of a way to avoid this at the moment. At least we are aware of it.